### PR TITLE
Create ticket management model and interface for AB#14513

### PR DESCRIPTION
**Pull request by BrianMaki**
# Implements [AB#14513](https://dev.azure.com/qslvic/304a1f8c-dace-4f85-adf3-bf563d5b3a39/_workitems/edit/14513)

## Description

Create model and interface for Ticket Management system.

- Create Ticket model
- Create CheckInRequest model
- Create TicketStatus enum
- Create interface for createTicket, removeTicket and updateTicket 

## Testing

- [ ] Unit Tests Updated
- [ ] Functional Tests Updated
- [x] Not Required

## UI Changes

No

## Items to Review:

-   [General PR Guidelines](https://github.com/bcgov/healthgateway/wiki/PRguidance)

### CODE REVIEW

**Review by sslaws — APPROVED** [review]

File: Apps/WebClient/src/ClientApp/src/services/interfaces.ts [hunk 1]
```diff
@@ -41,6 +42,8 @@ import UserRating from "@/models/userRating";
 import VaccinationStatus from "@/models/vaccinationStatus";
 import { RootState } from "@/store/types";
 
+import { CheckInRequest } from "./../models/checkInRequest";
```
Comment (MikeLyttle) [inline Apps/WebClient/src/ClientApp/src/services/interfaces.ts:45]: The `@/models/` syntax is preferred for portability.

File: Apps/WebClient/src/ClientApp/src/services/interfaces.ts [hunk 2]
```diff
@@ -277,3 +280,9 @@ export interface ILogger {
 export interface IStoreProvider {
     getStore(): Store<RootState>;
 }
+
+export interface ITicketService {
+    createTicket(room: string): Ticket;
+    removeTicket(checkInRequest: CheckInRequest): void;
+    updateTicket(checkInRequest: CheckInRequest): Ticket;
```
Comment (MikeLyttle) [inline Apps/WebClient/src/ClientApp/src/services/interfaces.ts:287]: The return values should be wrapped in a Promise<>.

There should also be a definition for the initialize function:
```
initialize(config: ExternalConfiguration, http: IHttpDelegate): void;
```

**Review by MikeLyttle — CHANGES_REQUESTED** [review]

File: Apps/WebClient/src/ClientApp/src/constants/ticketStatus.ts [hunk 3]
```diff
@@ -0,0 +1,7 @@
+export const enum TicketStatus {
+    Queued = "Queued",
+    Processed = "Processed",
```
[the remark this answers is not in the document]
Comment (BrianMaki) [reply Apps/WebClient/src/ClientApp/src/constants/ticketStatus.ts:3]: Yes I will do this via AB#14508.

File: Apps/WebClient/src/ClientApp/src/services/interfaces.ts [hunk 4]
```diff
@@ -285,4 +284,8 @@ export interface ITicketService {
     createTicket(room: string): Promise<Ticket>;
     removeTicket(checkInRequest: CheckInRequest): Promise<void>;
     updateTicket(checkInRequest: CheckInRequest): Promise<Ticket>;
+    initialize(
+        config: ExternalConfiguration,
+        http: IHttpDelegate
+    ): Promise<void>;
```
Comment (MikeLyttle) [inline Apps/WebClient/src/ClientApp/src/services/interfaces.ts:290]: This one shouldn't return a promise.

**Review by MikeLyttle — CHANGES_REQUESTED** [review]

**Review by MikeLyttle — APPROVED** [review]

**Pull request merged**